### ISSUE TITLE
[bug] a-fold select outer fold when cursor is at startRow of fold

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#636]
When cursor is at 2nd line `a z`(`a-fold`)
 - It should select `inner-fold`(2nd to 3rd rows).
 - But it actually select `outer-fold`(1st to 3rd rows).

```coffeescript
describe "outer-fold", ->
  describe "inner-fold", ->
    # body
```

**Comment by t9md** [comment]
## How I fixed this

#### old behavior

When cursor is at foldStartRow
 - `i z`: Ignore fold at cursor Row
 - `a z`: Ignore fold at cursor Row

### new behavior

When cursor is at foldStartRow
 - `i z`: Ignore fold at cursor Row
 - `a z`: Select fold at cursor Row


## `i z` and `a z` is not consistent, but it's by intention.

See following code as example.

```coffeescript
   getRange: (selection) ->
     console.log "hello"

     if true
       console.log "world"
```

When cursor is at `if`-expression row

- A. `i f` select `outer-fold`(fold starting from `getRange:...`).
- B. `a f` select `cursor-fold`(fold starting from `if true`).
- C. `i f` consistent to `a f`, it select inner-area of `cursor-fold`(`comsole.log` row.), but it make difficult to select inner-area of `outer-fold` or feels unintuitive in many situatino.

In other words, `if` row have two aspect, `foldStartRow` and `justPartOfRowIncludedInOuterFold`.
So I treat this situation as like this

- `i f`: treat it as `justPartOfRowIncludedInOuterFold`.
- `a f`: treat it as `foldStartRow`.

**Comment by t9md** [comment]
# Final conclusion

- Old behavior
  - Cursor is at start of fold, both `i z` and `a z` **ignore** cursor fold.

- New behavior
  - Cursor is at start of fold, both `i z` and `a z` **not ignore** cursor fold.
